src/Parser/TypeParser.php

```diff
@@ -66,15 +75,29 @@ public function parse(TokenIterator $tokens): Ast\Type\TypeNode
 	 * @internal
 	 * @template T of Ast\Node
 	 * @param T $type
+	 * @param list<Ast\Comment> $comments
+	 *
 	 * @return T
 	 */
-	public function enrichWithAttributes(TokenIterator $tokens, Ast\Node $type, int $startLine, int $startIndex): Ast\Node
+	public function enrichWithAttributes(TokenIterator $tokens, Ast\Node $type, int $startLine, int $startIndex, array $comments = []): Ast\Node
 	{
 		if ($this->useLinesAttributes) {
 			$type->setAttribute(Ast\Attribute::START_LINE, $startLine);
 			$type->setAttribute(Ast\Attribute::END_LINE, $tokens->currentTokenLine());
 		}
 
+		if ($this->useCommentsAttributes) {
+			if ($tokens->currentTokenType() === Lexer::TOKEN_COMMENT) {
+				$comments[] = new Ast\Comment(
+					$tokens->currentTokenValue(),
+					$tokens->currentTokenLine(),
+					$tokens->currentTokenIndex()
+				);
+				$tokens->next();
```

Comment:
What if there's one comment after another? `/* c *//* d*/`

Reply:
fair enough: 5d9be65

Although I really don't know what it means, or how this all gets consumed by other libraries.